### PR TITLE
Some changes

### PR DESCRIPTION
**Pull request by ocordes**
Hi,

I've written this pull request, because we want to use the spell checking in a learning environment and your package still has some unimplemented feature and a small BUG. 

I guess the whole request is not useful, but please check a few changes. First there was a small BUG when hitting the Cancel-Button of the language picker. The value test was wrong and creates a unusable picker.

Then the code is now saving the language selection in the user settings and reading the setting back during activation. Also changing manually the setting works fine now. There is a simple test that only languages which are predefined are accepted otherwise the change will be ignored but only mentioned in the console.log. A dialog would be interesting, but not necessary since many will change the language with the picker.

For my own purpose I've added the german language de-de to the predefined languages. There are also more, but I want to implement something like special node-package(s) which contain separate language sets, because noboday needs all languages at the same time.

If you like, we can work together. I'm a programmer for several languages and also a python teacher and IT admin.

Cheers,
   Oliver

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ The extension has been tested up to JupyterLab version 2.2.0.
 ## Installation
 
 ```bash
-jupyter labextension install @ijmbarr/jupyterlab_spellchecker
+jupyter labextension install @ocordes/jupyterlab_spellchecker
```
Comment (krassowski) [inline README.md:17]: I guess this should stay unchanged.

File: src/index.ts [hunk 2]
```diff
@@ -134,8 +151,20 @@ class SpellChecker {
...
-        this.ignored_tokens = new Set(tokens)
+        this.ignored_tokens = new Set(tokens);
         this.accepted_types = settings.get('mimeTypes').composite as Array<string>;
+
+        // read the saved language setting
+        let language_code = settings.get('language').composite;
+        let user_language = languages.filter(l => l.code == language_code)[0];
+        if (user_language === undefined)
+        {
+          console.warn('The language '+language_code+' is not supportet!')
```
Comment (krassowski) [inline src/index.ts:162]:
```suggestion
          console.warn('The language ' + language_code + ' is not supported!')
```

File: src/index.ts [hunk 3]
```diff
@@ -448,7 +484,7 @@ function activate(app: JupyterFrontEnd, tracker: INotebookTracker, palette: ICom
  * Initialization data for the jupyterlab_spellchecker extension.
  */
 const extension: JupyterFrontEndPlugin<void> = {
-    id: '@ijmbarr/jupyterlab_spellchecker:plugin',
+    id: '@ocordes/jupyterlab_spellchecker:plugin',
```
Comment (krassowski) [inline src/index.ts:487]: not needed necessary in this PR I think

File: src/index.ts [hunk 4]
```diff
@@ -104,27 +108,40 @@ class SpellChecker {
...
               this.update_settings(settings);
               settings.changed.connect(() => {
                   this.update_settings(settings);
+
+                  // load the dictionary
+                  this.load_dictionary().catch(console.warn);
               });
+
+              // load the dictionary
+              this.load_dictionary().catch(console.warn);
```
Comment (krassowski) [inline src/index.ts:144]: Could `load_dictionary()` be moved into `update_settings()` instead of repeating it twice here?

File: src/index.ts [hunk 5]
```diff
@@ -408,11 +441,14 @@ class SpellChecker {
             title: 'Choose spellchecker language',
             items: choices.map(language => language.name)
         }).then(value => {
-            if (value != null) {
+            if (value.value != null) {
```
Comment (krassowski) [inline src/index.ts:444]: Good catch!

**Comment by krassowski** [comment]
Very nice! It would probably help if you could track down what is the license for the German dictionary (to make sure it can be redistributed) and where it came from, so that it can be updated in the future.

File: package.json [hunk 6]
```diff
@@ -1,18 +1,18 @@
...
   ],
-  "homepage": "https://github.com/ijmbarr/jupyterlab_spellchecker",
+  "homepage": "https://github.com/ocordes/jupyterlab_spellchecker",
   "bugs": {
-    "url": "https://github.com/ijmbarr/jupyterlab_spellchecker"
+    "url": "https://github.com/ocordes/jupyterlab_spellchecker"
   },
   "license": "BSD-3-Clause",
-  "author": "Iain Barr",
+  "author": "Iain Barr + Oliver Cordes",
```
Comment (krassowski) [inline package.json:15]: I think that package.json should remain unchanged, except for adding your name into `contributors` property (which needs to be created - an array of strings, or persons as defined in the [npmjs documentation](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#people-fields-author-contributors)) if you wish to have the recognition.

**Comment by krassowski** [comment]
> There are also more, but I want to implement something like special node-package(s) which contain separate language sets, because noboday needs all languages at the same time.

There are plans to handle that in the framework of language packs of JupyterLab 3.0 (pending JupyterLab 3.0 release), have a look here: https://github.com/jupyterlab/team-compass/issues/85. It would be great to see it happen! I am currently swamped with my studies but hope to catch up over the Christmas break. I would love to collaborate with you and we could kick off by moving this repo into the JupyterLab organization.

In other words, do not invest too much time into language packages now, because the new extensions system in JupyterLab 3.0 will render quite a few things obsolete and better do it then in coordination with others working on localization :) Thank you for this PR!

File: README.md (hunk 1 above)
Comment (ocordes) [reply README.md:17]: Yes.

File: package.json (hunk 6 above)
Comment (ocordes) [reply package.json:15]: Please, do what is best, I'm new to npmjs and this was a quick&dirty hack to get this running in my production enviroment.

**Review by ocordes — COMMENTED** [review]
The licence for the german package the apache licence. I've downloaded the files from https://chromium.googlesource.com/chromium/deps/hunspell_dictionaries/+/master ! As far as I was told by a colleague,  the files needs to be converted to use the german "Umlaute" "ä,ö,ü,ß" properly. 

Okay, I will not do any more work now ;-) I'm happy if I can join for this project. For my working projects we introduced JupyterLab for Science and parallel for teaching (Physics&Astronomy of the university of Bonn (Germany)) and the number of users is still growing. Actually, I didn't know that there were more plans in the background!

**Comment by ocordes** [comment]
BTW, I've checked some of the dictionaries for other languages and it seems that for a good internationalization typo-js has some problems. I have a working example for german, french, Spanish, Portuguese, but I cannot get the files working for italiano. The problem is that the list of words which is generated from the rules is maybe to long and the generating to time consuming that typo.js is not useful. I briefly checked the code and maybe someone should think of a better implementation of the hunspell algorithm in JS. Therefor I suggest to move also the typo.js repo into jupyter as well or do a integration into the spell checking repo to have a better control of the code ... Maybe it is worth to think about this, before moving ...

**Comment by ijmbarr** [comment]
Thanks for the changes!
